deterministic-bloom-wasm/src/lib.rs

```diff
@@ -27,29 +23,167 @@ pub fn set_panic_hook() {
     console_error_panic_hook::set_once();
 }
 
-#[wasm_bindgen]
-extern "C" {
-    // For alerting
-    pub(crate) fn alert(s: &str);
-    // For logging in the console.
-    #[wasm_bindgen(js_namespace = console)]
-    pub fn log(s: &str);
-}
-
 //------------------------------------------------------------------------------
 // Macros
 //------------------------------------------------------------------------------
 
-/// Return a representation of an object owned by JS.
-#[macro_export]
-macro_rules! value {
-    ($value:expr) => {
-        wasm_bindgen::JsValue::from($value)
+/// Generate a monomorphic [BloomFilter] newtypes.
+macro_rules! gen_bloom {
+    ($name: ident, $n: expr, $k: expr) => {
+        #[doc = concat!("A monomorphic wrapper for [BloomFilter]`<", stringify!($n), ", ", stringify!($k), ">`.")]
+        #[wasm_bindgen]
+        #[derive(Debug, Default, From, Into)]
+        pub struct $name {
+            pub(crate) boxed: Box<BloomFilter<$n, $k>>,
+        }
+
+        #[wasm_bindgen]
+        impl $name {
+            #[doc = concat!("Initialize a blank [", stringify!($name), "] (i.e. contains no elements).")]
```

Comment:
wish there was a better way than all these sub-concat! and stringify! macros, but I guess that's the ticket, sadly.